src/Subscribers/WatchdogSubscriber.php

```diff
@@ -105,58 +105,82 @@ public function watchdog(string $jid, string $worker, ?int $pid = null): void
+            foreach ($pubsub as $message) {
+                if ($message->kind !== 'message' || empty($message->payload)) {
+                    continue;
+                }
+
+                $payload = json_decode($message->payload, true);
+                if (empty($payload)) {
+                    continue;
+                }
+
+                if (empty($payload['event']) || !is_array($payload)) {
+                    continue;
+                }
+
+                if (!in_array($payload['event'], self::WATCHDOG_EVENTS, true) || empty($payload['jid'])) {
+                    continue;
+                }
+
+                if ($payload['jid'] !== $jid) {
+                    continue;
+                }
+
+                switch ($payload['event']) {
+                    case self::LOCK_LOST:
+                        if (!empty($payload['worker']) && $payload['worker'] === $worker) {
+                            $this->logger->info(
+                                "{type}: sending SIGKILL to child {$pid}; job {jid} handed out to another worker",
+                                [
+                                    'type' => 'watchdog:' . $worker,
+                                    'jid' => $jid,
+                                ]
+                            );
+
+                            $this->system->posixKill($pid, SIGKILL);
+                            $pubsub->stop();
+                        }
+                        break;
+                    case self::CANCELED:
+                        if (!empty($payload['worker']) && $payload['worker'] === $worker) {
+                            $this->logger->info(
+                                "{type}: sending SIGKILL to child {$pid}; job {jid} canceled",
+                                [
+                                    'type' => 'watchdog:' . $worker,
+                                    'jid' => $jid,
+                                ]
+                            );
+                            $this->system->posixKill($pid, SIGKILL);
+                            $pubsub->stop();
+                        }
+                        break;
+                    case self::COMPLETED:
+                    case self::FAILED:
                         $pubsub->stop();
-                    }
-                    break;
-                case self::COMPLETED:
-                case self::FAILED:
-                    $pubsub->stop();
-                    break;
+                        break;
+                }
             }
+        } catch (\Throwable $exception) {
+            $this->logger->critical(
+                "Critical error",
+                [
+                    'type' => 'watchdog:' . $worker,
+                    'jid' => $jid,
+                    'exception' => $exception
+                ]
+            );
+            $this->logger->info(
+                "{type}: sending SIGKILL to child {$pid}; job {jid} canceled due to exception",
```

Comment:
Тут, мабуть, так повинно бути?
```suggestion
                "{$type}: sending SIGKILL to child {$pid}; job {$jid} canceled due to exception",
```

Reply:
там не змінні. в інших логах теж так.  власне звідти і копіював.